### DISCUSSION TITLE
WSL cannot use Httpie to connect to "localhost:8080"

### DISCUSSION THREAD

**Discussion by NickSanguansuk** [microsoft/WSL#7083] [Q&A]
I run my Spring Boot app in IntelliJ (Tomcat started on ports: 8080).
I don't know why I cannot access my webApp using Httpie on WSL. (I can access my webApp using Httpie on PowerShell just fine.)

I can access it with browser ---> localhost:8080
I can access it with PowerShell ---> http localhost:8080   or   curl.exe localhost:8080

But I **cannot** access it with WSL ---> http localhost:8080
The error is below:

http: error: ConnectionError: HTTPConnectionPool(host='localhost', port=8080): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fd4a37da3a0>: Failed to establish a new connection: [Errno 111] Connection refused')) while doing GET request to URL: http://localhost:8080/

---------- ---------- ---------- ----------

However, the WSL can use Httpie to access google.com just fine (but cannot access the localhost).
In WSL:
$ http google.com           ---> Working fine.
$ http localhost:8080       ---> Error

Is it because the IntelliJ is in Windows, not WSL?

**Comment by natnael-bedru** [comment]
You can work around this by getting the windows ip via `ipconfig /all` and using the ip instead of **localhost**

http 192.168.1.2:8080
